packages/kuma-gui/src/app/connections/views/ConnectionsXdsConfigView.vue

```diff
@@ -20,39 +20,48 @@
       <XCard>
         <DataLoader
           :src="uri(sources, `/connections/xds/for/:proxyType/:name/:mesh/:endpoints`, {
-            proxyType: ({ ingresses: 'zone-ingress', egresses: 'zone-egress'})[route.params.proxyType] ?? 'dataplane',
+            proxyType: ({ ingresses: 'zone-ingress', egresses: 'zone-egress' })[route.params.proxyType] ?? 'dataplane',
             name: route.params.proxy,
             mesh: route.params.mesh || '*',
             endpoints: String(route.params.includeEds),
           })"
           v-slot="{ data, refresh }"
         >
-          <XCodeBlock
-            language="json"
-            :code="JSON.stringify(data, null, 2)"
-            is-searchable
-            :query="route.params.codeSearch"
-            :is-filter-mode="route.params.codeFilter"
-            :is-reg-exp-mode="route.params.codeRegExp"
-            @query-change="route.update({ codeSearch: $event })"
-            @filter-mode-change="route.update({ codeFilter: $event })"
-            @reg-exp-mode-change="route.update({ codeRegExp: $event })"
+          <XWindow
+            :resize="true"
+            v-slot="{ resize }"
           >
-            <template #primary-actions>
-              <XCheckbox
-                :checked="route.params.includeEds"
-                label="Include Endpoints"
-                @change="(value) => route.update({ includeEds: value})"
-              />
-              <XAction
-                action="refresh"
-                appearance="primary"
-                @click="refresh"
+            <div
+              ref="$el"
+            >
```

Comment:
Bit annoyed about this, the return of meaningless divs.

It seems in order to use a ref to get a reference deep in your component tree (in this case *View.vue > XCodeBlock > XLayout), you have to do a load of `defineExpose` plumbing, which is just going to be brittle and hard to maintain, and at the very least annoying to have to use so much code to get to a deep DOM element.

Instead I just figured I could just wrap the thing in an extra div and get the position of the extra div instead (we need the `.top` of the XCodeBlock to calculate the height it should be)